beacon-chain/blockchain/receive_attestation.go

```diff
@@ -89,7 +89,7 @@ func (s *Service) ReceiveAttestationNoPubsub(ctx context.Context, att *ethpb.Att
 
 	// Skip checking for competing attestation's target roots at epoch boundary.
 	if !helpers.IsEpochStart(attSlot) {
-		targetRoot, err := helpers.BlockRoot(s.headState, att.Data.Target.Epoch)
+		targetRoot, err := helpers.BlockRoot(s.HeadState(), att.Data.Target.Epoch)
```

Comment:
lets not clone the state here, we are only reading from the state

Reply:
fixed